internal/users/db/migration.go

```diff
@@ -174,6 +187,7 @@ var schemaMigrations = []schemaMigration{
 			}
 
 			query := `UPDATE users SET name = LOWER(name);
+					  UPDATE groups SET ugid = LOWER(ugid) WHERE ugid = name;
 					  UPDATE groups SET name = LOWER(name);`
```

Comment:
This will fail if there are any groups with names that only differ in case. https://github.com/ubuntu/authd/issues/789 indicates that this might actually happen, because it's also possible that users are in multiple groups with the exact same display name. It's not clear to me how these groups are added, because the when I try to add a group via the Entra admin center with a name that already exists or only differs in case, it won't let me add the group, but multiple users reported that issue, so that can definitely happen.

The result will be that the query fails, causing authd to start to fail. To avoid that, I propose we delete all groups that would cause the migration to fail.

Reply:
Ok, that's an option for me... Better than failing the login IMHO

Reply:
with the brokers we currently support, it should not be possible to have multiple users with names that only differ in case. should we still also check user names and delete users which would cause the migration to fail?

Reply:
Probably not a big deal no... If you want to do it just in case, feel free.

Reply:
then I'll not do that